src/Common/tests/Cryptography/AlgorithmImplementations/ECDsa/ECDsaTests.cs

```diff
@@ -0,0 +1,473 @@
+        {
+            Assert.Throws<ArgumentNullException>("data",
+                () => ecdsa.VerifyData((Stream)null, null, default(HashAlgorithmName)));
+        }
+
+        [Theory, MemberData("AllImplementations")]
+        public void VerifyDataStream_NullSignature_ThrowsArgumentNullException(ECDsa ecdsa)
+        {
+            Assert.Throws<ArgumentNullException>("signature",
+                () => ecdsa.VerifyData(new MemoryStream(), null, default(HashAlgorithmName)));
+        }
+
+        [Theory, MemberData("AllImplementations")]
+        public void VerifyDataStream_DefaultHashAlgorithm_ThrowsArgumentException(ECDsa ecdsa)
+        {
+            Assert.Throws<ArgumentException>("hashAlgorithm",
+                () => ecdsa.VerifyData(new MemoryStream(), new byte[0], default(HashAlgorithmName)));
+        }
+
+        //////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+        [Fact]
+        public void SignDataByteArray_BaseWithValidArguments_ThrowsNotImplementedBecauseHashDataOverrideIsNeeded()
+        {
+            Assert.Throws<NotImplementedException>(
+                () => new ECDsaStub().SignData(new byte[0], HashAlgorithmName.SHA384));
+        }
+        [Fact]
+        public void SignDataByteArraySpan_BaseWithValidArguments_ThrowsNotImplementedBecauseHashDataOverrideIsNeeded()
+        {
+            Assert.Throws<NotImplementedException>(
+                () => new ECDsaStub().SignData(new byte[0], 0, 0, HashAlgorithmName.SHA384));
+        }
+
+        [Fact]
+        public void SignDataStream_BaseWithValidArguments_ThrowsNotImplementedBecauseHashDataOverrideIsNeeded()
+        {
+            Assert.Throws<NotImplementedException>(
+                () => new ECDsaStub().SignData(new MemoryStream(), HashAlgorithmName.SHA512));
+        }
+
+        [Fact]
+        public void VerifyDataByteArray_BaseWithValidArguments_ThrowsNotImplementedBecauseHashDataOverrideIsNeeded()
+        {
+            Assert.Throws<NotImplementedException>(
+                () => new ECDsaStub().VerifyData(new byte[0], new byte[0], HashAlgorithmName.SHA384));
+        }
+
+        [Fact]
+        public void VerifyDataStream_BaseWithValidArguments_ThrowsNotImplementedBecauseHashDataOverrideIsNeeded()
+        {
+            Assert.Throws<NotImplementedException>(
+                () => new ECDsaStub().VerifyData(new MemoryStream(), new byte[0], HashAlgorithmName.SHA384));
+        }
+
+
+        //////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+        
+        public static IEnumerable<object[]> RealImplementations()
+        {
+            return new[] { 
+                new ECDsa[] { ECDsaFactory.Create() },
+            };
+        }
+
+        [Theory, MemberData("RealImplementations")]
+        public void SignHash_NullHash_ThrowsArgumentNullException(ECDsa ecdsa)
```

Comment:
I could be wrong, but I don't really see a situation where we'd have more than one implementation type per library; so it feels to me like the Theory-ness can be squished down to [Fact] and the variable replaced with a factory call.

But it might not be worth the trouble.